systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java

```diff
@@ -366,6 +366,34 @@ void testKafkaMirrorMaker2WrongBootstrap() {
         KafkaMirrorMaker2Resource.deleteKafkaMirrorMaker2WithoutWait(kafkaMirrorMaker2);
     }
 
+    @Test
+    void testKafkaTopicDecreaseStatus() {
+        KafkaTopicResource.topic(CLUSTER_NAME, TEST_TOPIC_NAME, 5).done();
+        int decreaseTo = 1;
+
+        LOGGER.info("Decreasing number of partitions to {}", decreaseTo);
+        KafkaTopicResource.replaceTopicResource(TEST_TOPIC_NAME, kafkaTopic -> kafkaTopic.getSpec().setPartitions(decreaseTo));
+        KafkaTopicUtils.waitForKafkaTopicPartitionChange(TEST_TOPIC_NAME, decreaseTo);
+
+        assertKafkaTopicDecreasePartitionsStatus(TEST_TOPIC_NAME);
```

Comment:
The original issue wasn't just that the status was wrong, but also that it flipped back to `Ready` on the next reconciliation. So if this is intended as a regression test we should probably wait for a reconciliation interval and make the same assertion again.

Reply:
Yes, you are right, I didn't read the issue correctly, thanks for the comment.